src/main/java/com/arpnetworking/http/Routes.java

```diff
@@ -156,22 +156,29 @@ public Routes(
                                 .withStatus(StatusCodes.OK)
                                 .withEntity(JSON_CONTENT_TYPE, ByteString.fromString(STATUS_JSON)));
             }
-        } else if ((HttpMethods.POST.equals(request.method()))
-                && path.equals(COLLECTD_V1_SOURCE_PREFIX)) {
-            final Future<ActorRef> refFuture = _actorSystem.actorSelection("/user/collectdv1")
-                    .resolveOne(FiniteDuration.create(1, TimeUnit.SECONDS));
-            return FutureConverters.toJava(refFuture).thenCompose(
-                    ref -> {
-                        final CompletableFuture<HttpResponse> response = new CompletableFuture<>();
-                        ref.tell(new RequestReply(request, response), ActorRef.noSender());
-                        return response;
-                    })
-                    .exceptionally(err -> HttpResponse.create().withStatus(404));
+        } else if (HttpMethods.POST.equals(request.method())) {
+            if (path.equals(COLLECTD_V1_SOURCE_PREFIX) || path.equals("/source/collectd")) {
+                return dispatchHttpRequest(request, "/user/collectdv1");
+            } else if (path.equals(APP_V1_SOURCE_PREFIX)) {
+                return dispatchHttpRequest(request, "/user/appv1");
+            }
         }
 
         return CompletableFuture.completedFuture(HttpResponse.create().withStatus(404));
     }
 
+    private CompletionStage<HttpResponse> dispatchHttpRequest(final HttpRequest request, final String actorName) {
+        final Future<ActorRef> refFuture = _actorSystem.actorSelection(actorName)
+                .resolveOne(FiniteDuration.create(1, TimeUnit.SECONDS));
+        return FutureConverters.toJava(refFuture).thenCompose(
+                ref -> {
+                    final CompletableFuture<HttpResponse> response = new CompletableFuture<>();
+                    ref.tell(new RequestReply(request, response), ActorRef.noSender());
+                    return response;
+                })
+                .exceptionally(err -> HttpResponse.create().withStatus(404));
```

Comment:
This is partially true.  The actor is only started if there is a pipeline that starts it up.  So, it may be there or it may not.  We should talk about how pipelines work in the new world.